src/api/router.rs

```diff
@@ -184,14 +221,15 @@ pub fn book() -> BoxedFilter<(impl Reply,)> {
             .and(warp::path::param::<u32>())
             .and(warp::path::end())
             .and(warp::get())
+            .and(with_db_state(db_state.clone()))
             .and_then(super::endpoints::book::designations_retrieve)
             )
 
         // GET - /book/tags
         .or(warp::path("tags")
             .and(warp::path::end())
             .and(warp::get())
-            .and(warp::query())
```

Comment:
I think this is needed for pagination.

Reply:
can you please elaborate what you mean by pagination? As per docs `query` just deserializes the query to the defined type

Reply:
Sorry, my bad. I've added it to all the endpoints, because I thought we would also have a pagination feature on them. I assume some of the lists to become quite long. To implement the page and limit you need the query parameters.

Reply:
Do you mean that in case of pagination you need the query as json to be deserialized to some page struct? I guess it's more relevant to the /book/ endpoint and I've left the `query` there. I don't think we have too many designations to have on /book/designations and others, except maybe for tags. But still do we need pagination for those? I don't see really see the use case for that.

Reply:
I'm not sure if we mean the same thing. I meant query parameters like"...&page=1&limit=20" to be serialized in a struct, so that you can use them in the function. You are right that it's not required right now for tags, but other endpoints already have quite some data (languages, authors, publishers). I will accept the PR, because it's not an issue right now. But it might be worth implementing it, also to be consistent over all the endpoints.

Reply:
I see now, yeah you're right it definitely makes sense to use it for languages, authors and publishers too. I'll put it back when I'll get to implementing those. It makes no difference right now anyway